core/src/labels/labels.cpp

```diff
@@ -58,15 +58,17 @@ void Labels::processLabelUpdate(const ViewState& viewState,
         ScreenTransform transform { m_transforms, transformRange };
 
         // Use extendedBounds when labels take part in collision detection.
-        auto bounds = (onlyTransitions || !label->canOcclude())
+        auto bounds = (onlyRender || !label->canOcclude())
             ? screenBounds
             : extendedBounds;
 
         if (!label->update(mvp, viewState, &bounds, transform)) {
             continue;
         }
 
-        if (onlyTransitions) {
+        if (label->forceInvisible()) { label->occlude(); }
```

Comment:
Not really if you don't want it to be included in the visible set you wouldn't need to occlude it, what could also work is to simply to put it in the `dead` state when the marker is set to not visible.